pandas/core/arrays/sparse/array.py

```diff
@@ -1666,8 +1667,9 @@ def make_sparse(
     return sparsified_values, index, fill_value
 
 
-def make_sparse_index(length, indices, kind):
+def make_sparse_index(length, indices, kind) -> SparseIndex:
 
+    index: SparseIndex
```

Comment:
This could probably be narrowed further in a followup using the relation
`kind = "integer" -> returns IntIndex`
`kind = "block" -> returns BlockIndex`